lib/krb5/expand_path.c

```diff
@@ -449,11 +476,20 @@ _expand_token(krb5_context context,
 	    return _expand_extra_token(context, p[1], ret);
     }
 
-    for (i = 0; i < sizeof(tokens)/sizeof(tokens[0]); i++) {
-	if (!strncmp(token+2, tokens[i].tok, (token_end - token) - 2))
-	    return tokens[i].exp_func(context, tokens[i].param,
-				      tokens[i].postfix, ret);
-    }
+    colon = strchr(token+2, ':');
```

Comment:
This might run past `token_end`,  in which case we should again use `token_end`.

Reply:
Yeah, this is actually worse because I really should not use strchr(3) as I have no guarantee that I'll find either a nul or a colon before I SEGV. Please find the updated open coded strchr(3)-like code.